File: src/android/input/HandleNotificationInput.java
package com.bandyer.cordova.plugin.input;

import org.json.JSONArray;
import org.json.JSONObject;

import com.bandyer.cordova.plugin.Constants;

public class HandleNotificationInput {

    private String mPayload;

    public static HandleNotificationInput createFrom(JSONArray argsArray) throws PluginInputNotValidException {
        HandleNotificationInput handleNotificationInput = new HandleNotificationInput();
        try {
            JSONObject args = (JSONObject) argsArray.get(0);
            String payload = args.getString(Constants.ARG_HANDLE_NOTIFICATION);
            if(payload == null || payload.equals("")) {
                throw new PluginInputNotValidException(Constants.ARG_HANDLE_NOTIFICATION + " cannot be null");
            }
            handleNotificationInput.setPayload(payload);

            return handleNotificationInput;
        } catch (Throwable t) {
            throw new PluginInputNotValidException("error on InitInput", t);
        }
    }

    public String getPayload() {
        return mPayload;
    }

    public void setPayload(String payload) {
        mPayload = payload;
    }
}

File: src/android/cordova_plugin/src/main/java/com/bandyer/cordova/plugin/input/HandleNotificationInput.kt
package com.bandyer.cordova.plugin.input

import org.json.JSONArray
import org.json.JSONObject

import com.bandyer.cordova.plugin.Constants

class HandleNotificationInput {

    var payload: String? = null

    companion object {

        @Throws(PluginInputNotValidException::class)
        fun createFrom(argsArray: JSONArray): HandleNotificationInput {
            val handleNotificationInput = HandleNotificationInput()
            try {
                val args = argsArray.get(0) as JSONObject
                val payload = args.getString(Constants.ARG_HANDLE_NOTIFICATION)
                if (payload == null || payload == "") {
                    throw PluginInputNotValidException(Constants.ARG_HANDLE_NOTIFICATION + " cannot be null")
                }
                handleNotificationInput.payload = payload

                return handleNotificationInput
            } catch (t: Throwable) {
                throw PluginInputNotValidException("error on InitInput", t)
            }

        }
    }
}
